Documentation/advanced-configuration.md

```diff
@@ -0,0 +1,318 @@
+```
+
+Then to delete a pool, `rbd` in this example, run:
+```bash
+ceph osd pool rm rbd rbd --yes-i-really-really-mean-it
+```
+
+### Creating a pool
+```bash
+ceph osd pool create rbd 2000 2000 replicated replicated_ruleset
+```
+
+`replicated_ruleset` is the default CRUSH rule that replicates between the hosts and OSDs in the `default` root hierarchy.  Visit [this page](http://docs.ceph.com/docs/kraken/rados/operations/placement-groups/) for more information on how to decide the number of PGs for your pools.
+
+### Setting the number of replicas
+The `size` setting of a pool tells the cluster how many copies of the data should be kept for redundancy.  By default the cluster will distribute these copies between `host` buckets in the CRUSH Map.  This can be set when [creating a pool via ThirdPartyResource](https://github.com/rook/rook/blob/master/Documentation/pool-tpr.md) or after creation with `ceph`.
+
+So for example let's change the `size` of the `rbd` pool to three
+```
+ceph osd pool set rbd size 3
+```
+
+Now if you run `ceph -s` or `rook status` you may see "recovery" operations and PGs in "undersized" and other "unclean" states.  The cluster is essentially fixing itself since the number of replicas has been increased, and should go back to "active/clean" state shortly, after data has been replicated between hosts.  When that's done you will be able to lose 2/3 of your storage nodes and still have access to all your data in that pool.  Of course you will only have 1/3 the capacity as a tradeoff.
+
+## Custom ceph.conf settings
+With Rook the full swath of [Ceph settings](http://docs.ceph.com/docs/kraken/rados/configuration/) are available to use on your storage cluster.  When we supply Rook with a ceph.conf file those settings will be propagated to all Mon, OSD, etc daemons to use.
+
+In this example we will set the default pool `size` to three, and tell OSD daemons not to change the weight of OSDs on startup.
+
+Here's our custom ceph.conf:
+```bash
+[global]
+osd crush update on start = false
+osd pool default size     = 2
+```
+
+### Standalone Rook
+With our ceph.conf file created we can pass the settings to `rookd` with the `--ceph-config-override` argument, for example:
+```bash
+rookd --ceph-config-override=ceph.conf --data-devices=sdf
+```
+
+### Kubernetes
+With rook-operator on Kubernetes we can edit the OSD DaemonSet after it has been created.  This will be streamlined once [#571](https://github.com/rook/rook/issues/571) is resolved.
+
+#### Using a ConfigMap
+First we will create a ConfigMap that will hold our custom ceph.conf, so it can be mounted as a file to be consumed in our Rook pods.
+```bash
+# Create a ConfigMap from a text file
+kubectl -n rook create configmap rook-config --from-file ceph.conf
+
+# View the resulting ConfigMap
+kubectl -n rook get configmap rook-config -o yaml
+
+# Edit the ConfigMap within the Kubernetes cluster
+kubectl -n rook edit configmap rook-config
```

Comment:
this step isn't necessary unless they want to edit the config map later on, right?  if they are happy with their original ceph.conf file that the config map was created from, they can skip this?  if so, perhaps we can make that more obvious, because a lot of people just copy/paste blindly.